Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift

```diff
@@ -182,6 +182,15 @@ private extension Sequence<DeclarationRenderSection.Token> {
 // MARK: ExternalEntity
 
 private extension LinkDestinationSummary {
+    /// A value that indicates whether this symbol is under development and likely to change.
```

Comment:
Looks like you're adding exactly the same code to both `LinkDestinationSummary` and `OutOfProcessReferenceResolver.ResolvedInformation` - would it be worth introducing a protocol?

Actually I just saw [this comment](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/External%20Data/OutOfProcessReferenceResolver.swift#L548):

```
    public struct ResolvedInformation: Codable {
        // This type is duplicating the information from LinkDestinationSummary with some minor differences.
        // Changes generally need to be made in both places. It would be good to replace this with LinkDestinationSummary.
        // FIXME: https://github.com/apple/swift-docc/issues/802
```

Is this a good time to take on this refactoring work?

Reply:
Yes that comment was exactly my reasoning for duplicating the API as closely as possible, thanks for calling that out! 

I'll look into refactoring the code to see how much of a code change it would be to unify the code paths.

Reply:
The change to unify the code paths appears to be quite involved, I'll look into it further tomorrow but I think we should tackle that as a separate PR regardless.

Reply:
That's definitely for a separate PR. It's would both be quite involved and a bit of a risky change.